components/renku_data_services/base_models/core.py

```diff
@@ -51,17 +51,16 @@ class GitlabAPIUser(APIUser):
     name: str
     gitlab_url: str
 
-    @property
-    def gitlab_graphql_url(self):
-        """Gets the graphql url for gitlab."""
+    def __post_init__(self):
+        """Sets the graphql url for gitlab."""
         gitlab_url = self.gitlab_url
 
         if not gitlab_url.startswith("http") and "://" not in gitlab_url:
-            gitlab_url = f"https://{gitlab_url}"
+            raise errors.ConfigurationError(message=f"Gitlab URL should start with 'http(s)://', got: {gitlab_url}")
 
         gitlab_url = gitlab_url.rstrip("/")
 
-        return f"{gitlab_url}/api/graphql"
+        self.gitlab_url = f"{gitlab_url}/api/graphql"
```

Comment:
I think we should be more specific here. This should be called `self.gitlab_graphql_url`. As a user of this class without reading the code I would expect that `self.gitlab_url` is the base url of gitlab. And that is what it used to be prior to this change. So this could lead to confusion and problems down the line.